src/bootstrap/src/core/config/config.rs

```diff
@@ -2335,6 +2319,29 @@ impl Config {
                 None => None,
                 Some(commit) => {
                     self.download_ci_rustc(commit);
+
+                    let builder_config_path =
+                        self.out.join(self.build.triple).join("ci-rustc/builder-config");
+                    let ci_config_toml = Self::get_toml(&builder_config_path);
+                    let current_config_toml = Self::get_toml(self.config.as_ref().unwrap());
```

Comment:
It's a big edge case, but if you don't have `config.toml` and just do `./x build --set rust.download_rustc=true`, this will crash.